src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -41,6 +42,77 @@ def persist_flow_run(self, run_info: FlowRunInfo):
         self.queue.put(run_info)
 
 
+class LineProcessManager:
```

Comment:
The class name is not appropriate.
Consider rename to HealthyEnsuredProcess
Otherwise the method like `LineProcessManager.put` is confusing in meaning.
Rename "start_new_process" to "start_new", "end_process" to "end"